modules/custom/az_search_api/src/Plugin/search_api/processor/AZImageDerivative.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace Drupal\az_search_api\Plugin\search_api\processor;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\PluginFormInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\image\ImageStyleInterface;
+use Drupal\search_api\Attribute\SearchApiProcessor;
+use Drupal\search_api\Plugin\PluginFormTrait;
+use Drupal\search_api\Processor\FieldsProcessorPluginBase;
+use Drupal\search_api\Utility\Utility;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Creates absolute links to image derivatives.
+ */
+#[SearchApiProcessor(
+  id: 'az_image_derivative',
+  label: new TranslatableMarkup('Quickstart Image Derivative'),
+  description: new TranslatableMarkup('Transform a file URI into an image derivative.'),
+  stages: [
+    'preprocess_index' => -20,
+  ],
+)]
+class AZImageDerivative extends FieldsProcessorPluginBase implements PluginFormInterface {
+
+  use PluginFormTrait;
+
+  /**
+   * The entity type manager.
+   */
+  protected ?EntityTypeManagerInterface $entityTypeManager = NULL;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    /** @var static $processor */
+    $processor = parent::create($container, $configuration, $plugin_id, $plugin_definition);
+    $processor->setEntityTypeManager($container->get('entity_type.manager'));
+    return $processor;
+  }
+
+  /**
+   * Retrieves the entity type manager.
+   *
+   * @return \Drupal\Core\Entity\EntityTypeManagerInterface
+   *   The entity type manager.
+   */
+  public function getEntityTypeManager(): EntityTypeManagerInterface {
+    return $this->entityTypeManager ?: \Drupal::entityTypeManager();
+  }
+
+  /**
+   * Sets the entity type manager.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   *
+   * @return $this
+   */
+  public function setEntityTypeManager(EntityTypeManagerInterface $entity_type_manager): static {
```

Comment:
This is a new pattern I haven't seen before for entityTypeManager.

Reply:
I think this is modeled on how the parent `FieldsProcessorPluginBase` plugin class that this extends handles its dependencies (right @tadean?).